tests/vmi_cloudinit_test.go

```diff
@@ -539,47 +535,44 @@ var _ = Describe("[rfe_id:151][crit:high][vendor:cnv-qe@redhat.com][level:compon
 
 				By("checking cloudinit network-config")
 				CheckCloudInitFile(vmi, "openstack/latest/network_data.json", testNetworkData)
+				CheckCloudInitFile(vmi, "openstack/latest/user_data", testUserData)
 
 				// Expect that the secret is not present on the vmi itself
 				vmi, err = virtClient.VirtualMachineInstance(testsuite.GetTestNamespace(vmi)).Get(context.Background(), vmi.Name, metav1.GetOptions{})
 				Expect(err).ToNot(HaveOccurred())
-				Expect(vmi.Spec.Volumes[idx].CloudInitConfigDrive.NetworkData).To(BeEmpty())
-				Expect(vmi.Spec.Volumes[idx].CloudInitConfigDrive.NetworkDataBase64).To(BeEmpty())
+				for _, volume := range vmi.Spec.Volumes {
+					if volume.CloudInitConfigDrive != nil {
+						Expect(volume.CloudInitConfigDrive.NetworkData).To(BeEmpty())
```

Comment:
Also check for userdata to be empty and for secret refs to be present?

Reply:
I think this is out of slope of this pr. We can do that on a follow up pr, but honestly, I'm not sure it's so important.

Reply:
At least check for userdata to be empty to be consistent with the test below?

Reply:
done